### ISSUE TITLE
Invariant Violation: eventType argument is required.

### ISSUE THREAD

**Issue by sultan-arshi** [react-navigation/react-navigation#9475]
**Current Behavior**

- What code are you running and what is happening?
```
import { useNavigation } from "@react-navigation/native"
const { goBack } = useNavigation()
```
**and onPress event is** 
`onPress={() => goBack()}`

- Include a screenshot or video if it makes sense.

![Simulator Screen Shot - iPhone 11 - 2021-03-31 at 10 39 36](https://user-images.githubusercontent.com/18505536/113116090-6912c500-920d-11eb-8cc9-2c86c4ce813d.png)

**How to reproduce**
i just checked the ios version and it is not working
Basically, i have made `BackButton` to reuse it.
and i am using only single `StackNavigator` which contain all of my screesns
i am also surprised why it is not working on some screen and on some screen it working 

library info:
"@react-navigation/native": "^5.9.3",

**Comment by sultan-arshi** [comment]
> The versions mentioned in the issue for the following packages differ from the latest versions on npm:
> 
> * `@react-navigation/stack` (found: `5.0.0-alpha.37`, latest: `5.14.3`)
> 
> Can you verify that the issue still exists after upgrading to the latest versions of these packages?

 issue still exists

**Comment by sultan-arshi** [comment]
it was not due to the `goBack()` or navigation issue 
i was using some native modules and adding/subscribing an event, for removing i was using `eventEmitter.removeAllListeners()` which was the cause of crash 
i just replaced  it with `eventEmitter.removeListener('CustomeEventName', () => {})`
